### PR TITLE
multiple builds

### LINKED ISSUE #259: Release multiple binaries
Add a `binaries` or something like that to the config file.

The idea is in [this comment](https://github.com/goreleaser/goreleaser/issues/49#issuecomment-300814646) by @hellais:

> This is currently the limiting fact for me using goreleaser for my releases. I have a single package where I would like to ship multiple binaries that each are generated from a separate main.go. I would like all the binaries to end up inside of the final .tar.gz so having separate goreleaser.yml files is not an option.
> 
> Could this issue be re-opened or should I file another one?

### PR DESCRIPTION
**Pull request by caarlos0**
Now we can have several build, several binaries even, using the `builds`  key on the `goreleaser.yml` file!

```yaml
builds:
- binary: blah
  main: ./blah/main.go
  # etc
```

It is backwards compatible with the previous `build` field, and in the end it will releases a single archive per target (goos + goarch + goarm) and also a single brew formula, with all binaries of that given target inside.

It required some rework in the code, but, after a few hours, I was able to tackle it down and also fix some unknown/unreported bugs on the way! ( #271  )

I tested this PR on a test repo: 

- https://github.com/caarlos0/test/releases/tag/v3.3.0
- https://github.com/caarlos0/test/blob/master/goreleaser.yml
- https://github.com/caarlos0/homebrew-tap/blob/master/Formula/aaaaaaaaa.rb

Everything looks good: non darwin/amd64 builds where ignored by the brew formulae, deb files are working, etc...

---

FWIW, here are snapshots of the latest commit:

[goreleaser_Darwin_i386.tar.gz](https://github.com/goreleaser/goreleaser/files/1118068/goreleaser_Darwin_i386.tar.gz)
[goreleaser_Darwin_x86_64.tar.gz](https://github.com/goreleaser/goreleaser/files/1118069/goreleaser_Darwin_x86_64.tar.gz)
[goreleaser_Linux_arm64.tar.gz](https://github.com/goreleaser/goreleaser/files/1118070/goreleaser_Linux_arm64.tar.gz)
[goreleaser_Linux_armv6.tar.gz](https://github.com/goreleaser/goreleaser/files/1118071/goreleaser_Linux_armv6.tar.gz)
[goreleaser_Linux_i386.tar.gz](https://github.com/goreleaser/goreleaser/files/1118072/goreleaser_Linux_i386.tar.gz)
[goreleaser_Linux_x86_64.tar.gz](https://github.com/goreleaser/goreleaser/files/1118073/goreleaser_Linux_x86_64.tar.gz)
[goreleaser_checksums.txt](https://github.com/goreleaser/goreleaser/files/1118074/goreleaser_checksums.txt)
[goreleaser_Windows_i386.zip](https://github.com/goreleaser/goreleaser/files/1118075/goreleaser_Windows_i386.zip)
[goreleaser_Windows_x86_64.zip](https://github.com/goreleaser/goreleaser/files/1118076/goreleaser_Windows_x86_64.zip)

---

closes #259 
refs #49 #228

### CODE REVIEW

File: pipeline/brew/brew.go [hunk 1]
```diff
@@ -136,30 +137,24 @@ func doBuildFormula(data templateData) (bytes.Buffer, error) {
...
+		Name:         formulaNameFor(ctx.Config.Name),
 		Desc:         ctx.Config.Brew.Description,
 		Homepage:     ctx.Config.Brew.Homepage,
 		Repo:         ctx.Config.Release.GitHub,
 		Tag:          ctx.Git.CurrentTag,
 		Version:      ctx.Version,
-		Binary:       ctx.Config.Build.Binary,
 		Caveats:      ctx.Config.Brew.Caveats,
 		File:         file,
-		Format:       ctx.Config.Archive.Format, // TODO this can be broken by format_overrides
```
Comment (caarlos0) [inline pipeline/brew/brew.go]: this was finally fixed on a3c1e2b

**Comment by caarlos0** [comment]
@jorinvo @client9 can you guys review?

It should work now!

I already tested in a test repo: https://github.com/caarlos0/test/releases/tag/v3.3.0

Everything looks good: non darwin/amd64 builds where ignored by the brew formulae, deb files are working, etc...

BTW: I think it is a good idea to add a pre-step to remove the `dist` folder, maybe with a switch or something... because now I can have multiple binaries and etc, and changing the config may affect the end result by including a previous run build. 

What do you guys think?

**Comment by hellais** [comment]
Thanks for working on this! I will check it out.

File: README.md [hunk 2]
```diff
@@ -61,36 +61,36 @@ By default GoReleaser will build the current directory, but you can change the b
...
-  goarch:
-    - amd64
+builds:
+  - binary: drum-roll
+    goos:
+      - windows
+      - darwin
+      - linux
+    goarch:
+      - amd64
```
Comment (hellais) [inline README.md]: Is the legacy `build` syntax also supported? If so maybe it's worth mentioning that it's also still supported.

File: pipeline/build/build_test.go [hunk 3]
```diff
@@ -193,26 +208,28 @@ func TestRunInvalidLdflags(t *testing.T) {
...
 	}
 	t.Run("pre-hook", func(t *testing.T) {
-		ctx.Config.Build.Hooks.Pre = "exit 1"
+		ctx.Config.Builds[0].Hooks.Pre = "exit 1"
 		assert.Error(Pipe{}.Run(ctx))
 	})
 	t.Run("post-hook", func(t *testing.T) {
-		ctx.Config.Build.Hooks.Post = "exit 1"
+		ctx.Config.Builds[0].Hooks.Post = "exit 1"
 		assert.Error(Pipe{}.Run(ctx))
```
Comment (hellais) [inline pipeline/build/build_test.go]: Again if the legacy `Build` syntax is supported, you may want to leave the unittests for those too to ensure backward compatibility is preserved.

File: README.md (hunk 2 above)
Comment (caarlos0) [inline README.md]: it is supported but is "deprecated"...

File: pipeline/build/build_test.go (hunk 3 above)
Comment (caarlos0) [inline pipeline/build/build_test.go]: good idea.

Comment (caarlos0) [inline pipeline/build/build_test.go]: actually, this is being tested already!

https://github.com/goreleaser/goreleaser/blob/master/pipeline/defaults/defaults_test.go#L75

the single build will always become an item of the builds list :)

**Pull request merged**